uvicorn/protocols/websockets/wsproto_impl.py

```diff
@@ -159,25 +201,31 @@ def handle_bytes(self, event):
 
     def handle_close(self, event):
         self.queue.put_nowait({"type": "websocket.disconnect", "code": event.code})
+        output = self.conn.send(event.response())
+        self.transport.write(output)
         self.transport.close()
 
     def handle_ping(self, event):
-        output = self.conn.bytes_to_send()
+        # output = self.conn.bytes_to_send()
+        output = self.conn.send(event.response())
         self.transport.write(output)
 
     def send_500_response(self):
         headers = [
             (b"content-type", b"text/plain; charset=utf-8"),
             (b"connection", b"close"),
         ]
-        msg = h11.Response(
+        if self.conn.connection is None:
+            output = self.conn.send(wsproto.events.RejectConnection(status_code=500))
+        else:
```

Comment:
Cannot send below stuff during the handshake. I suspect this `if self.conn.connection is None` check should be done in a few other places (maybe where `_upgrade_connection` was previously used?).